DataLayer/Entities/Group.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace GameBoard.DataLayer.Entities
+{
+    public class Group
+    {
+        public int Id { get; set; }
+
+        public string OwnerId { get; set; }
+        public ApplicationUser Owner { get; set; }
+
+        public string Name { get; set; }
+
+        public ICollection<GroupUser> GroupUser { get; set; }
+
+        public ICollection<ApplicationUser> Users
+        {
+            get => GroupUser?.Select(gu => gu.User).ToList();
+            set => GroupUser = value.Select(
+                u => new GroupUser
+                {
+                    Group = this,
+                    User = u
+                }).ToList();
+        }
```

Comment:
I remember looking at this and saying it was OK.
But it's not.
First off, calling this property causes the `GroupUser` collection to be effectively copied, which is a performance hit.
That wouldn't be that bad, but consider what happens when someone goes:
`group.Users.Add(user);`
Clearly you would expect the user to be added as the group's user. But here it won't do anything. `Users` returns a copied collection that is immediately marked for garbage collection after this line. `GroupUsers` will never be updated.
My idea would be to make `Users` a get-only property returning an `IEnumerable`, and create a method `AddUser(ApplicationUser)`. At the very least make sure we do not copy collections in properties (it's an actual bad practice to do that).

Reply:
Marking as done.